### PR TITLE
Removed unmaintained psyco dependency.

### PR DESCRIPTION
**Pull request by tanaydin**
Added requirements.txt for package installation
Fixed code for pycodestyle checking (with --max-line-length=120)
Tests extended
Help string formatted
Version changed.

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -2,20 +2,20 @@
...
 
-for n in range(1, 1001):
-	prod = 1
-	for factor in factors(n, False, False, 7.97308847044, 1.0):
-		prod *= factor
-	if prod != n:
-		print('Failed to factor %d.' % n, file=sys.stderr)
-		sys.exit(1)
+for n in range(1, 2047):
+    prod = 1
```
Comment (martingkelly) [inline test:15]: Please no extraneous whitespace changes; the original code had tabs, so this should have tabs too.

File: test [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
-from pyecm import *
 
-for n in range(1, 1001):
-	prod = 1
-	for factor in factors(n, False, False, 7.97308847044, 1.0):
-		prod *= factor
-	if prod != n:
-		print('Failed to factor %d.' % n, file=sys.stderr)
-		sys.exit(1)
+for n in range(1, 2047):
```
Comment (martingkelly) [inline test:14]: Is there any reason to believe that this will expose more bugs than the original value? I have no particular issue with it but am not sure the change is necessary either.

File: pyecm.py [hunk 3]
```diff
@@ -1,1483 +1,1569 @@
-#!/usr/bin/python3
-
-'''
-You should install psyco and gmpy if you want maximal speed.
+#!/usr/bin/python
```
Comment (martingkelly) [inline pyecm.py:1]: This should stay as `python3`, as many Linux distros still default to `python` --> `python2`.

File: pyecm.py [hunk 4]
```diff
@@ -1,1483 +1,1569 @@
...
+        """
+        Computes the Greatest Common Divisor of a and b using the standard quadratic time
+        improvement to the Euclidean Algorithm. Returns the GCD of a and b.
+        """
 
-      if b == 0:
-         return a
-      elif a == 0:
-         return b
+        if number2 == 0:
```
Comment (martingkelly) [inline pyecm.py:47]: (1) As before, please no extraneous whitespace changes; here an unnecessary space is added. This applies to the rest of the file as well.

(2) I prefer `a` and `b` to `number1` and `number2`, as they are standard notation for GCD, and the new names convey no additional information.

File: pyecm.py [hunk 5]
```diff
@@ -1,1483 +1,1569 @@
...
+                if n <= 3:
+                    return 3 - (n <= 2)
+                n += (n & 1) ^ 1
+                return n + (((4 - (n % 3)) >> 1) & 2)
 
-         p = 1
-         for i in range(int(math.log(n) / LOG_2), 0, -1):
-            p <<= (n >> i) & 1
-            p = (p * p) % n
+            n += (n & 1) ^ 1
```
Comment (martingkelly) [inline pyecm.py:151]: Why did you remove the for loop here?

File: pyecm.py [hunk 6]
```diff
@@ -1,1483 +1,1569 @@
...
 MULT = math.log(3) / LOG_2
 ONE = mpz(1)
-SMALL = 2.0**(-30)
+SMALL = 2.0 ** (-30)
 SMALLEST_COUNTEREXAMPLE_FASTPRIME = 2047
 T = (type(mpz(1)), type(1), type(1))
-DUMMY = 'dummy' # Dummy value throughout the program
-VERSION = '2.0.5 (Python 3)'
+DUMMY = 'dummy'  # Dummy value throughout the program
+VERSION = '3.0.0'
```
Comment (martingkelly) [inline pyecm.py:239]: Please no version changes here. Version changes occur when I choose to do a release, not on every commit. Plus, there's no reason to bump the major version here.

File: pyecm.py [hunk 7]
```diff
@@ -1,1483 +1,1569 @@
...
+"""
 Filename: pyecm
 Authors: Eric Larson <elarson3@uoregon.edu>, Martin Kelly <martin@martingkelly.com>, Matt Ford <zeotherm@gmail.com>
 License: GNU GPL (see <http://www.gnu.org/licenses/gpl.html> for more information.
 Description: Factors a number using the Elliptic Curve Method, a fast algorithm for numbers < 50 digits.
 
-We are using curves in Suyama's parametrization, but points are in affine coordinates, and the curve is in Wierstrass form.
-The idea is to do many curves in parallel to take advantage of batch inversion algorithms. This gives asymptotically 7 modular multiplications per bit.
+We are using curves in Suyama's parametrization, but points are in affine coordinates,
+and the curve is in Wierstrass form.
```
Comment (martingkelly) [inline pyecm.py:10]: Line 11 should join with line 10, wrapping to the max of 80 characters.

**Review by martingkelly — CHANGES_REQUESTED** [review]
Thanks for the PR! Please separate into the following commits:

(a) Remove psyco dependency
(b) Add requirements.txt
(c) Formatting changes, with no version change, no variable renaming, no extraneous whitespace changes.
(d) Any other changes you'd like to make

In this way, I can merge (a) and (b) while we discuss (c) and (d) further.

File: pyecm.py (hunk 5 above)
Comment (tanaydin) [reply pyecm.py:151]: It's still there, in 180. line.

Comment (martingkelly) [reply pyecm.py:151]: OK; github's diffs make it hard to tell!
